client/src/pages/Balance.js

```diff
@@ -16,16 +17,25 @@ import AddIcon from "@material-ui/icons/Add";
 import RemoveIcon from "@material-ui/icons/Remove";
 
 const BalancePage = ({ state, dispatch }) => {
+  const priorTopUp = state.cart.find((item) => item.name === "Review Credits")
+    ?.quantity;
   const user = useContext(userContext);
-  const [topUp, setTopUp] = useState(2);
+  const [topUp, setTopUp] = useState(() => priorTopUp || 2);
+  const history = useHistory();
 
   const handleClick = ({ target, ...e }) => {
     const { id } = target.closest("button");
-    if (id === "incr") return setTopUp(topUp + 1);
+    if (id === "incr") return setTopUp(() => topUp + 1);
     if (id === "decr" && topUp > 0) return setTopUp(topUp - 1);
   };
 
-  const handleCheckout = (e) => {};
+  const handleCheckout = (e) => {
+    dispatch({
+      type: "ADD_ITEM_TO_CART",
+      item: { name: "Review Credits", quantity: topUp, unitCost: 10 },
```

Comment:
Can we make details about this item as constants somewhere - i.e. the name, the unit cost and the initial value? Something in some sort of constants / utils file you can reference

Reply:
Totally, I'm assuming at a certain point there will be BE data to request for this.